nixos/modules/security/doas.nix

```diff
@@ -0,0 +1,255 @@
+              '';
+            };
+
+            setEnv = mkOption {
+              type = with types; listOf str;
+              default = [];
+              description = ''
+                Keep or set the specified variables. Variables may also be removed with a leading
+                ‘-’ or set using <code>variable=value</code>. If the first character of value is a
+                ‘$’ then the value to be set is taken from the existing environment variable of the
+                indicated name. This option is processed after the default environment has been
+                created.
+              '';
+            };
+
+            users = mkOption {
+              type = with types; listOf (either str int);
+              default = [];
+              description = "The usernames / UIDs this rule should apply for.";
+            };
+
+            groups = mkOption {
+              type = with types; listOf (either str int);
+              default = [];
+              description = "The groups / GIDs this rule should apply for.";
+            };
+
+            runAs = mkOption {
+              type = with types; nullOr str;
+              default = null;
+              description = ''
+                Under which user/group the specified command is allowed to run.
+
+                A user can be specified using just the username: <code>"foo"</code>. It is also
+                possible to only allow running as a specific group with <code>":bar"</code>.
+              '';
+            };
+
+            cmd = mkOption {
+              type = with types; nullOr str;
+              default = null;
+              description = ''
+                The command the user is allowed to run. The default is all commands. Be advised that
+                it is best to specify absolute paths. If a relative path is specified, only a
+                restricted PATH will be searched. When set to <code>null</code>, all commands are
+                allowed.
+              '';
+            };
+
+            args = mkOption {
+              type = with types; nullOr (listOf str);
+              default = [];
+              description = ''
+                Arguments to the command. The arguments provided by the user need to match those
+                specified. When set to <code>null</code>, the command must be run without any
+                arguments.
+              '';
+            };
+          };
+        }
+      );
+    };
+
+    extraConfig = mkOption {
+      type = with types; lines;
+      default = "";
+      description = ''
+        Extra configuration text appended to <filename>doas.conf</filename>.
+      '';
+    };
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    security.doas.extraRules = [
+      {
+        groups = [ "wheel" ];
+        noPass = !cfg.wheelNeedsPassword;
+      }
+    ];
+
+    # TODO setenv SSH_AUTH_SOCK?
+    security.doas.configFile = ''
```

Comment:
Forgot to change this. Fixed soon.

Reply:
And to address that TODO just above, I've opted to add `setenv { SSH_AUTH_SOCK }` to every rule by default, which closely follows what `sudo`'s module does (though in a less-pretty manner). Still open for debate on this approach, though.